File: PYTHON/student.py
# student.py
# Student mewarisi Person

from person import Person

class Student(Person):
    def __init__(self, id="", nama="", usia=0, nim="", jurusan="", gpa=0.0):
        super().__init__(id, nama, usia)
        self.nim = nim
        self.jurusan = jurusan
        self.gpa = gpa

    def get_nim(self): return self.nim
    def set_nim(self, nim): self.nim = nim

    def get_jurusan(self): return self.jurusan
    def set_jurusan(self, jurusan): self.jurusan = jurusan

    def get_gpa(self): return self.gpa
    def set_gpa(self, gpa): self.gpa = gpa

File: JAVA/Student.java
// Student.java
// Student mewarisi Person (hierarchical inheritance)

public class Student extends Person {
    private String nim;
    private String jurusan;
    private double gpa;

    public Student() {
        super();
        this.nim = "";
        this.jurusan = "";
        this.gpa = 0.0;
    }

    public Student(String id, String nama, int usia, String nim, String jurusan, double gpa) {
        super(id, nama, usia);
        this.nim = nim;
        this.jurusan = jurusan;
        this.gpa = gpa;
    }

    // Getter
    public String getNim() { return nim; }
    public String getJurusan() { return jurusan; }
    public double getGpa() { return gpa; }

    // Setter
    public void setNim(String nim) { this.nim = nim; }
    public void setJurusan(String jurusan) { this.jurusan = jurusan; }
    public void setGpa(double gpa) { this.gpa = gpa; }
}